main.go

```diff
@@ -0,0 +1,258 @@
+		}
+		for _, channel := range allBundleChannels.List() {
+			if channel == pkg.defaultChannel {
+				fmt.Fprintf(os.Stdout, "\n"+indent2+"subgraph "+channel+" channel - default")
+			} else {
+				fmt.Fprintf(os.Stdout, "\n"+indent2+"subgraph "+channel+" channel")
+			}
+			replaceSet := sets.NewString()
+			skipRangeReplaceSet := sets.NewString()
+			for _, bundle := range pkg.bundles {
+				if bundle.channels.Has(channel) {
+					// if no replaces edges, just write the node
+					if bundle.replaces.Len() == 0 && bundle.skipRanges.Len() == 0 {
+						if bundle.channelHeads.Has(bundle.name + channel) {
+							replaceSet.Insert(bundle.name + "-" + channel + "(" + bundle.version + "):::head")
+						} else {
+							replaceSet.Insert(bundle.name + "-" + channel + "(" + bundle.version + ")")
+						}
+					}
+					for _, replace := range bundle.replaces.List() {
+						if bundle.skips == "" {
+							if bundle.channelHeads.Has(bundle.name + channel) {
+								replaceSet.Insert(replace + "-" + channel + "(" +
+									pkg.bundles[replace].version + ")" + " --> " + bundle.name + "-" +
+									channel + "(" + bundle.version + "):::head")
+							} else {
+								replaceSet.Insert(replace + "-" + channel + "(" +
+									pkg.bundles[replace].version + ")" + " --> " + bundle.name + "-" +
+									channel + "(" + bundle.version + ")")
+							}
+						} else {
+							if bundle.channelHeads.Has(bundle.name + channel) {
+								replaceSet.Insert(replace + "-" + channel + "(" +
+									pkg.bundles[replace].version + ")" + " x--x | " + bundle.skips + " | " + bundle.name + "-" +
+									channel + "(" + bundle.version + "):::head")
+							} else {
+								replaceSet.Insert(replace + "-" + channel + "(" +
+									pkg.bundles[replace].version + ")" + " x--x | " + bundle.skips + " | " + bundle.name + "-" +
+									channel + "(" + bundle.version + ")")
+							}
+						}
+					} // end bundle replaces edge graphing
+					for _, skipRangeReplace := range bundle.skipRanges.List() {
+						if bundle.channelHeads.Has(bundle.name + channel) {
+							skipRangeReplaceSet.Insert(skipRangeReplace + "-" + channel +
+								"(" + pkg.bundles[skipRangeReplace].version + ")" + " o--o | " + bundle.skipRange + " | " +
+								bundle.name + "-" + channel + "(" + bundle.version + "):::head")
+						} else {
+							skipRangeReplaceSet.Insert(skipRangeReplace + "-" + channel +
+								"(" + pkg.bundles[skipRangeReplace].version + ")" + " o--o | " + bundle.skipRange + " | " +
+								bundle.name + "-" + channel + "(" + bundle.version + ")")
+						}
+					} // end bundle skipRanges edge graphing
+				}
+			}
+			// reconcile items that appear in both sets to decide which kind of edge to draw
+			for _, replaceLine := range replaceSet.List() {
+				fmt.Fprintf(os.Stdout, "\n"+indent3+replaceLine)
+			}
+			for _, skipRangeReplaceLine := range skipRangeReplaceSet.List() {
+				fmt.Fprintf(os.Stdout, "\n"+indent3+skipRangeReplaceLine)
+			}
```

Comment:
WDYT about extracting the code here in funcs for we are able to read that better?

Reply:
Makes sense. Done in ee7f4cc5